chain/chain/src/store/mod.rs

```diff
@@ -210,17 +210,18 @@ pub trait ChainStoreAccess {
     fn get_incoming_receipts_for_shard(
         &self,
         epoch_manager: &dyn EpochManagerAdapter,
-        mut shard_id: ShardId,
+        target_shard_id: ShardId,
+        target_shard_layout: &ShardLayout,
         mut block_hash: CryptoHash,
         last_chunk_height_included: BlockHeight,
     ) -> Result<Vec<ReceiptProofResponse>, Error> {
         let _span =
-            tracing::debug_span!(target: "chain", "get_incoming_receipts_for_shard", ?shard_id, ?block_hash, last_chunk_height_included).entered();
+            tracing::debug_span!(target: "chain", "get_incoming_receipts_for_shard", ?target_shard_id, ?block_hash, last_chunk_height_included).entered();
 
         let mut ret = vec![];
 
-        let target_shard_id = shard_id;
-        let target_shard_layout = epoch_manager.get_shard_layout_from_prev_block(&block_hash)?;
+        let mut current_shard_id = target_shard_id;
+        let mut current_shard_layout = target_shard_layout.clone();
```

Comment:
Nice, maybe do the same for block hash? `let mut current_block_hash = block_hash` ?